pytorch_lightning/utilities/data.py

```diff
@@ -218,17 +226,29 @@ def _get_dataloader_init_kwargs(
     has_variadic_kwargs = any(p.kind is p.VAR_KEYWORD for p in params.values())
     if has_variadic_kwargs:
         # if the signature takes **kwargs, assume they will be passed down with `super().__init__(**kwargs)`
-        params.update(inspect.signature(DataLoader.__init__).parameters)
-        del params["self"]
+
+        # if the dataloader was wrapped in a hook, only take arguments with default values
+        #  and assume user passes their kwargs correctly
+        params.update(
+            {
+                k: v
+                for k, v in inspect.signature(DataLoader.__init__).parameters.items()
+                if not was_wrapped or (v.default is not v.empty)
```

Comment:
was_wrapped is invariant to this loop so it should go outside.

Reply:
You are right. During development, I had the condition outside of this dict comprehension and then I thought it looked neater this way.